### PR TITLE
BaSP-TG-78: Projects Form & Validations

### PR DESCRIPTION
**Pull request by DamianPalavecino**
Ticket: https://trello.com/c/WpnBg5Hl/78-form-projects

- Implement Input component
- Implement Joi validations
- Implement React hook forms
- Create a file named validations.jsx inside Form folder and create the schema using joi inside
- Use joiResolver() from React hook forms
- Replace relative paths with absolute paths
- Rename Projects.jsx and projectsForm.jsx files to index.jsx
- Create 'Form' folder to maintain consistency with the rest of the entities components
- Fix problem in Admin routes file with Project Form route

**Validations**:

- Project Name: 
Minimum 3 characters and 50 characters maximum, incluiding letters, numbers, spaces and symbols
Cannot be empty
Required
- Client Name:
Minimum 3 characters and 50 characters maximum, incluiding letters, numbers, spaces and symbols
Cannot be empty
Required
- Description:
Minimum 3 characters and 50 characters maximum, incluiding letters, numbers, spaces and symbols
Cannot be empty
Required
- Start Date:
Required
Cannot be empty
-  End Date: 
Required
Cannot be empty
Cannot be greater than Start Date
- Status: 
Required (on edit)

### CODE REVIEW

File: src/Components/Projects/Form/validations.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import Joi from 'joi';
+
+export const projectSchema = Joi.object({
+  name: Joi.string().min(3).max(50).required().messages({
+    'string.empty': 'Project Name cannot be empty',
```
Comment (Maxi-SR) [inline src/Components/Projects/Form/validations.jsx:5]: Use commas in all validations messages, like` ' "Project Name" cannot be empty" '`

File: src/Components/Projects/Form/index.jsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+import { joiResolver } from '@hookform/resolvers/joi';
+import { Input, Spinner, Button, Modal, Select } from 'Components/Shared';
+import { projectSchema } from './validations';
+
+const ProjectForm = () => {
+  const history = useHistory();
+  const params = useParams();
+  const dispatch = useDispatch();
+  const { message, isPending, list: projectData } = useSelector((state) => state.projects);
+  const id = params.id;
```
Comment (Maxi-SR) [inline src/Components/Projects/Form/index.jsx:16]: You can do this : `const { id } = useParams();`

File: src/Components/Projects/Form/index.jsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  const [showModal, setShowModal] = useState({ success: false, error: false });
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+    reset
+  } = useForm({
+    mode: 'onBlur',
```
[the remark this answers is not in the document]
Comment (DamianPalavecino) [reply src/Components/Projects/Form/index.jsx:24]: Done

File: src/Components/Projects/Form/validations.jsx (hunk 1 above)
Comment (DamianPalavecino) [reply src/Components/Projects/Form/validations.jsx:5]: Done

File: src/Components/Projects/Form/index.jsx (hunk 2 above)
Comment (DamianPalavecino) [reply src/Components/Projects/Form/index.jsx:16]: Done

File: src/Components/Projects/Form/validations.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+export const projectSchema = Joi.object({
+  name: Joi.string().min(3).max(50).required().messages({
+    'string.empty': '"Project Name" cannot be empty',
+    'string.min': '"Project Name" must have a minimum 3 characters',
+    'string.max': '"Project Name" must have a maximum 50 characters',
+    'any.required': '"Project Name" is required'
+  }),
+  startDate: Joi.date().max('now').required(),
+  endDate: Joi.date().max('now').required(),
```
Comment (Juanoauil) [inline src/Components/Projects/Form/validations.jsx:11]: these validations are correct?

Comment (DamianPalavecino) [reply src/Components/Projects/Form/validations.jsx]: No, they are outdated. I am going to update them after I merge the hotfix on the server repository

**Review by Juanoauil — APPROVED** [review]

**Pull request merged**